### ISSUE TITLE
How can I use the gradients of variable 'a' to update variable 'b'? 

### ISSUE THREAD

**Issue by HandsomeRocketboy** [tensorflow/tensorflow#24401]
Hi, I have two sets of weights 'a' and 'b' in my model, I want to use the gradients of 'a' to update 'b'. Can I call "compute_gradients" on 'a' to get the gradients, and use the result to call "apply_gradients" for 'b' to update weight 'b'? If feasible , what should I do?

**Comment by franneck94** [comment]
Does a and b has the same shape?

**Comment by HandsomeRocketboy** [comment]
> Does a and b has the same shape?

Yes, it does. In fact, I want to transfer resnet50 to a binary net, so there are two sets of weights in my model. One is a set of weight in float, named "w", the other is a set of binary weights, named "bin_w", just like this:

`    def binarize(self,x):
        gb = tf.get_default_graph()
        with ops.name_scope("binarized") as name:
            with gb.gradient_override_map({"Sign":"Identity"}):
                return tf.sign(x)

    def HardTanh(self,x):
        with tf.variable_scope("HardTanh"):
            return tf.clip_by_value(x,-1,1)

    def bin_conv(self,x,num_out_layers,kernel_size,stride=1,name=None,activation_fn="HardTanh"):
        #num_in_layers=x.get_shape().as_list()[3]
        with tf.variable_scope(name,reuse=False):
            p = np.floor((kernel_size - 1) / 2).astype(np.int32)
            p_x = tf.pad(x, [[0, 0], [p, p], [p, p], [0, 0]])
            num_in_layers=p_x.get_shape().as_list()[3]
            my_w=tf.get_variable('weight',[kernel_size,kernel_size,num_in_layers,num_out_layers],initializer=tf.contrib.layers.xavier_initializer_conv2d())
            my_w=tf.clip_by_value(my_w,-1,1)
            
            bin_w=self.binarize(my_w)
            bin_x=self.binarize(p_x)

            out=tf.nn.conv2d(bin_x,bin_w,strides=[1,stride,stride,1],padding='VALID')

            my_b=tf.get_variable("bias",[num_out_layers],initializer=tf.zeros_initializer)

            out=tf.nn.bias_add(out,my_b)

            ##########################################################
            out=tf.contrib.layers.batch_norm(out)
            ##########################################################

            if (activation_fn == None):
                return out
            else:
                return self.HardTanh(out)`

Now I want to use the gradients of 'bin_w' to update 'w', and binarize new 'w' to update 'bin_w', and take bin_w as the final result in my checkpoint file.
